tensorboard/webapp/metrics/views/main_view/card_grid_test.ts

```diff
@@ -0,0 +1,199 @@
+import {ScrollingModule} from '@angular/cdk/scrolling';
+import {Component, Input, NO_ERRORS_SCHEMA} from '@angular/core';
+import {
+  discardPeriodicTasks,
+  fakeAsync,
+  TestBed,
+  tick,
+} from '@angular/core/testing';
+import {PluginType} from '../../data_source';
+import {By} from '@angular/platform-browser';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {CardObserver} from '../card_renderer/card_lazy_loader';
+
+import {CardGridComponent} from './card_grid_component';
+import {CardGridContainer} from './card_grid_container';
+import {CardIdWithMetadata} from '../metrics_view_types';
+import {Store} from '@ngrx/store';
+import {MockStore, provideMockStore} from '@ngrx/store/testing';
+import {appStateFromMetricsState} from '../../testing';
+import {State} from '../../../app_state';
+import {selectors as settingsSelectors} from '../../../settings';
+import * as selectors from '../../../selectors';
+import {getMetricsTagGroupExpansionState} from '../../../selectors';
+
+const scrollElementHeight = 100;
+
+@Component({
+  selector: 'testing-component',
+  template: `
+    <div cdkScrollable>
+      <div class="placeholder">placeholder</div>
+      <metrics-card-grid
+        [cardIdsWithMetadata]="cardIdsWithMetadata"
+        [cardObserver]="cardObserver"
+        [groupName]="cardObserver"
+      ></metrics-card-grid>
+      <div class="placeholder">placeholder</div>
+    </div>
+  `,
+  styles: [
+    `
+      div {
+        position: fixed;
+        height: ${scrollElementHeight}px;
+        overflow-y: scroll;
+      }
+      .placeholder {
+        position: relative;
+        height: 700px;
+      }
+    `,
+  ],
+})
+class TestableComponent {
+  @Input() groupName: string = 'test group name';
+  @Input() cardIdsWithMetadata: CardIdWithMetadata[] = [];
```

Comment:
Do groupName and cardObserver actually need to be inputs? I don't see them used in the tests.

Reply:
Not needed. Removed.